VRDR/IJEMortality.cs

```diff
@@ -2491,21 +2492,57 @@ public string IDOB_YR
             }
         }
 
-        /// <summary>Infant Death/Birth Linking - year of birth</summary>
+        /// <summary>Infant Death/Birth Linking - Birth state</summary>
         [IJEField(90, 671, 2, "Infant Death/Birth Linking - State, U.S. Territory or Canadian Province of Birth - code", "BSTATE", 1)]
         public string BSTATE
         {
             get
             {
                 if (!String.IsNullOrWhiteSpace(BCNO))
                 {
-                    return Dictionary_Geo_Get("BSTATE", "PlaceOfBirth", "address", "state", true);
+                    String state = Dictionary_Get_Full("BSTATE", "BirthRecordState", "code");
+                    String retState;
+                // If the country is US or CA, strip the prefix
+                    if (state.StartsWith("US-") || state.StartsWith("CA-"))
+                    {
+                        // Console.Error.WriteLine("Starts with US- state= " + state + "retState =" + state.Substring(2));
+                        retState = state.Substring(3);
+                    } else {
+                        // Console.Error.WriteLine("No Prefix state= " + state);
+                        retState = state;
+                    }
+                    // Console.Error.WriteLine("BSTATE.get returning " + retState + ",  state = " + state);
+                    return retState;
                 }
                 return ""; // Blank
             }
             set
             {
-                // NOOP
+                if (!String.IsNullOrWhiteSpace(value))
+                {
+                    String birthCountry = BPLACE_CNT;
+                    String ISO31662code;
+                    Console.Error.WriteLine("BSTATE.set with country = " + birthCountry);
+                    switch (value){
+                        case "ZZ": // UNKNOWN OR BLANK U.S. STATE OR TERRITORY OR UNKNOWN CANADIAN PROVINCE OR UNKNOWN/ UNCLASSIFIABLE COUNTRY
+                            // Console.Error.WriteLine("BSTATE.set with value ZZ");
+                            return;  // do nothing 
+                        case "XX": //UNKNOWN STATE WHERE COUNTRY IS KNOWN, BUT NOT U.S. OR CANADA 
+                            // Console.Error.WriteLine("BSTATE.set with value XX");
+                            ISO31662code = birthCountry;
+                            break;
+                        default:  // a 2 character state
+                            if (birthCountry.Equals("US") || birthCountry.Equals("CA")){
+                                // Console.Error.WriteLine("BSTATE.set with US or CA" );
+                                ISO31662code = birthCountry + "-" + value;
+                            }else{
+                                ISO31662code = value;
+                            }
+                            // Console.Error.WriteLine("BSTATE.set with value " + ISO31662code );
+                            break;
```

Comment:
This seems like it opens up for the possibility of the birthCountry getting out of sync with the BirthRecordState, if the country is updated is the BSTATE also fixed to have the proper birthCountry?